src/NJsonSchema/JsonSchema.cs

```diff
@@ -890,6 +890,34 @@ private static JsonObjectType ConvertStringToJsonObjectType(string value)
 
         private void Initialize()
         {
+            bool IsOutOfRange<T>(decimal num)
+            {
+                try
+                {
+                    Convert.ChangeType(num, typeof(T));
+                    return false;
+                }
+                catch (OverflowException)
+                {
+                    return true;
+                }
+            }
+
+            if (Minimum.HasValue || Maximum.HasValue)
```

Comment:
@RicoSuter I'm not sure that this is the right place to put this logic as it is within the JsonSchema realm and is not specific to c# code generation.  My only other idea (but I don't like it too much) is to move this logic into the constructor of the NJsonSchema.CodeGeneration.CSharp.Models.PropertyModel class so that it modifies _property.ActualSchema.Format.  something like this:

```
            bool IsOutOfRange<T>(decimal num)
            {
                try
                {
                    System.Convert.ChangeType(num, typeof(T));
                    return false;
                }
                catch (System.OverflowException)
                {
                    return true;
                }
            }

            if (_property.ActualSchema.Minimum.HasValue || _property.ActualSchema.Maximum.HasValue)
            {
                if (string.IsNullOrEmpty(_property.ActualSchema.Format))
                {
                    if (_property.ActualSchema.Type == JsonObjectType.Integer)
                    {
                        // If an integer with no format is specified and then test Minimum and Maximum values are compatible with int32 and if not then explicitly set format to int64
                        if (IsOutOfRange<int>(_property.ActualSchema.Minimum.GetValueOrDefault()) || IsOutOfRange<int>(_property.ActualSchema.Maximum.GetValueOrDefault()))
                        {
                            _property.ActualSchema.Format = JsonFormatStrings.Long;
                        }
                    }
                }
            }

```

 Your thoughts?

Reply:
C# generator specific stuff must be moved into the generator model classes, e.g. in NJsonSchema.Generation or NJsonSchema.Generation.CSharp

Reply:
I have moved the code out of JsonSchema. Hopefully you can accept this PR now